### PR TITLE
Chore: Refactor `scroll-page.test.js` to TypeScript

### PR DESCRIPTION
**Pull request by CrypticSignal**
This PR is a refactor of `packages/jaeger-ui-components/src/scroll-page.test.js` to TypeScript.
#54027

### CODE REVIEW

File: packages/jaeger-ui-components/src/scroll-page.test.ts [hunk 1]
```diff
@@ -140,19 +144,19 @@ describe('scroll-by', () => {
...
+      expect(jest.mocked(window.scrollTo).mock.calls.length).toBe(1);
+      expect(jest.mocked(window.scrollTo).mock.calls[0][1]).toBe(value);
     });
 
     it('discards the in-progress scroll if the scroll is done', () => {
       // cause a `Tween` to be created to get a reference to _onTweenUpdate
       scrollTo(10);
       const { onUpdate, cancel: twCancel } = tweenInstances[0];
-      onUpdate({ value: 123, done: true });
+      onUpdate?.({ value: 123, done: true });
```
Comment (joey-grafana) [inline packages/jaeger-ui-components/src/scroll-page.test.ts:156]: Why the ?

File: packages/jaeger-ui-components/src/scroll-page.test.ts [hunk 2]
```diff
@@ -115,16 +116,19 @@ describe('scroll-by', () => {
...
+      expect(jest.mocked(tw.cancel).mock.calls.length).toBe(0);
     });
   });
 
   describe('_onTweenUpdate', () => {
-    let oldScrollTo;
+    let oldScrollTo: { (options?: ScrollToOptions | undefined): void; (x: number, y: number): void } & {
+      (options?: ScrollToOptions | undefined): void;
+      (x: number, y: number): void;
+    };
```
Comment (joey-grafana) [inline packages/jaeger-ui-components/src/scroll-page.test.ts:131]:
```suggestion
    let oldScrollTo: { (options?: ScrollToOptions | undefined): void; (x: number, y: number): void; };
```

File: packages/jaeger-ui-components/src/scroll-page.test.ts [hunk 3]
```diff
@@ -19,17 +19,18 @@ import { scrollBy, scrollTo, cancel } from './scroll-page';
...
-      const rv = { to: opts.to, onUpdate: opts.onUpdate };
+    jest.mocked(Tween).mockClear();
+    jest.mocked(Tween).mockImplementation((opts) => {
+      const rv = { to: opts.to, onUpdate: opts.onUpdate } as Tween;
       Object.keys(Tween.prototype).forEach((name) => {
         if (name !== 'constructor') {
+          // @ts-ignore
           rv[name] = jest.fn();
         }
       });
```
Comment (joey-grafana) [inline packages/jaeger-ui-components/src/scroll-page.test.ts:36]: The last time we added a //@ts-ignore it caused a bug!

Best not to add this unless completely necessary. Is this Object.keys even necessary?

File: packages/jaeger-ui-components/src/scroll-page.test.ts [hunk 4]
```diff
@@ -19,17 +19,18 @@ import { scrollBy, scrollTo, cancel } from './scroll-page';
...
 describe('scroll-by', () => {
   beforeEach(() => {
     window.scrollY = 100;
     tweenInstances.length = 0;
-    Tween.mockClear();
-    Tween.mockImplementation((opts) => {
-      const rv = { to: opts.to, onUpdate: opts.onUpdate };
+    jest.mocked(Tween).mockClear();
+    jest.mocked(Tween).mockImplementation((opts) => {
+      const rv = { to: opts.to, onUpdate: opts.onUpdate } as Tween;
```
Comment (joey-grafana) [inline packages/jaeger-ui-components/src/scroll-page.test.ts:30]:
```suggestion
      const rv = { to: opts.to, onUpdate: opts.onUpdate, cancel: jest.fn() };
```

**Review by joey-grafana — CHANGES_REQUESTED** [review]

File: packages/jaeger-ui-components/src/scroll-page.test.ts (hunk 1 above)
Comment (CrypticSignal) [reply packages/jaeger-ui-components/src/scroll-page.test.ts:156]: `onUpdate` can be `null` or `undefined`: https://github.com/grafana/grafana/pull/59617/files#diff-4afbe7ba43ccfac4cd8267a282c50a204ca048d775172b79d4e52993b6cc5214R37

Without a `?`, there will be a `Cannot invoke an object which is possibly 'null' or 'undefined'` error.

File: packages/jaeger-ui-components/src/scroll-page.test.ts (hunk 3 above)
Comment (CrypticSignal) [reply packages/jaeger-ui-components/src/scroll-page.test.ts]: This loop was adding `onUpdate` and `cancel` properties to the `rv` object and setting their values to `jest.fn()`. I have now removed this loop (so `//@ts-ignore` is no longer needed) and manually added `onUpdate` and `update`. See commit https://github.com/grafana/grafana/pull/59617/commits/9fd536bfda6cdf1125f0688c652d7a322c5f7d52.

File: packages/jaeger-ui-components/src/scroll-page.test.ts (hunk 4 above)
Comment (CrypticSignal) [reply packages/jaeger-ui-components/src/scroll-page.test.ts]: See https://github.com/grafana/grafana/pull/59617#discussion_r1041121133

File: packages/jaeger-ui-components/src/scroll-page.test.ts (hunk 2 above)
Comment (CrypticSignal) [reply packages/jaeger-ui-components/src/scroll-page.test.ts]: Done in commit https://github.com/grafana/grafana/pull/59617/commits/5630dd2f99edc0add1a20e31e871608f88803023.

**Review by joey-grafana — APPROVED** [review]

**Pull request merged**